spec/models/manageiq/providers/nuage/vsd_client/vcr_specs/vsd_client_spec.rb

```diff
@@ -0,0 +1,109 @@
+require 'json'
+describe ManageIQ::Providers::Nuage::NetworkManager::VsdClient do
+  before(:each) do
+    @userid = Rails.application.secrets.nuage_network.try(:[], 'userid') || 'NUAGE_USER_ID'
+    @password = Rails.application.secrets.nuage_network.try(:[], 'password') || 'NUAGE_PASSWORD'
+    @hostname = Rails.application.secrets.nuage_network.try(:[], 'host') || 'nuagenetworkhost'
+
+    # Ensure that VCR will obfuscate the basic auth
+    VCR.configure do |c|
+      # workaround for escaping host
+      c.before_playback do |interaction|
+        interaction.filter!(CGI.escape(@hostname), @hostname)
+        interaction.filter!(CGI.escape('NUAGE_NETWORK_HOST'), 'nuagenetworkhost')
+      end
+      c.filter_sensitive_data('NUAGE_NETWORK_AUTHORIZATION') { Base64.encode64("#{@userid}:#{@password}").chomp }
+    end
+  end
+
+  it "should return valid non empty json responses" do
+    VCR.use_cassette('manageiq/providers/nuage/network_manager/vsd_client/vsd_client', :allow_unused_http_interactions => true, :record => :new_episodes) do
+      vsd_client = ManageIQ::Providers::Nuage::NetworkManager::VsdClient.new("https://#{@hostname}:8443/nuage/api/v5_0", @userid, @password)
+
+      enterprises_resp = vsd_client.get_enterprises
+      domains_resp = vsd_client.get_domains
+      zones_resp = vsd_client.get_zones
+      subnets_resp = vsd_client.get_subnets
+      vports_resp = vsd_client.get_vports
+      vms_resp = vsd_client.get_vms
+      policy_groups_resp = vsd_client.get_policy_groups
+
+      assert_object_not_empty(enterprises_resp)
+      assert_json_object(enterprises_resp)
+      assert_common_parameters(enterprises_resp)
+
+      assert_object_not_empty(domains_resp)
+      assert_json_object(domains_resp)
+      assert_common_parameters(domains_resp)
+
+      assert_object_not_empty(zones_resp)
+      assert_json_object(zones_resp)
+      assert_common_parameters(zones_resp)
+
+      assert_object_not_empty(subnets_resp)
+      assert_json_object(subnets_resp)
+      assert_common_parameters(subnets_resp)
+
+      expect(vports_resp).to be true
+      assert_json_object(vports_resp)
+
+      assert_object_not_empty(vms_resp)
+      assert_json_object(vms_resp)
+      assert_common_parameters(vms_resp)
+
+      assert_object_not_empty(policy_groups_resp)
+      assert_json_object(policy_groups_resp)
+      assert_common_parameters(policy_groups_resp)
+    end
+  end
+
+  it "should fail on wrong password" do
+    VCR.use_cassette('manageiq/providers/nuage/network_manager/vsd_client/vsd_client_wrong_pass', :allow_unused_http_interactions => true, :record => :new_episodes) do
```

Comment:
`vsd_client/wrong_password` (shorten the path, and remove `vsd_client_` prefix since we're in `vsd_client` directory anyway). Also, remove the two arguments `:allow_unused_http_interactions => true, :record => :new_episodes`.